### PR TITLE
add plot_kwargs parameter and label encoder in confusion_matrix and AUC

### PR DESCRIPTION
**Pull request by moezali1**
…UC plot

## Related Issuse or bug

Info about Issue or bug

Closes #[issue number that will be closed through this PR]

#### Describe the changes you've made

A clear and concise description of what you have done to successfully close your assigned issue. Any new files? or anything you feel to let us know!

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)

A clear and concise description of it.

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Screenshots

 Original           | Updated
 :--------------------: |:--------------------:
 **original screenshot**  | <b>updated screenshot </b> |

### CODE REVIEW

File: pycaret/internal/tabular.py [hunk 1]
```diff
@@ -6643,11 +6659,21 @@ def confusion_matrix():
...
+            except:
+                encoder_dictionary = None
+
             visualizer = ConfusionMatrix(
                 pipeline_with_model,
                 random_state=seed,
                 fontsize=15,
                 cmap="Greens",
+                encoder=encoder_dictionary,
+                **plot_kwargs
```
Comment (Yard1) [inline pycaret/internal/tabular.py:6676]: shouldn't this be added to other visualizers as well?

File: pycaret/internal/tabular.py [hunk 2]
```diff
@@ -65,7 +65,7 @@
 import warnings
 from IPython.utils import io
 import traceback
-from unittest.mock import patch
+from unittest.mock import NonCallableMock, patch
```
Comment (Yard1) [inline pycaret/internal/tabular.py:68]: This isn't used, right?
```suggestion
from unittest.mock import patch
```

**Review by Yard1 — CHANGES_REQUESTED** [review]

File: pycaret/internal/tabular.py (hunk 1 above)
Comment (moezali1) [reply pycaret/internal/tabular.py:6676]: Yes. The PR is not complete yet. I will add it to all visualizers.

File: pycaret/internal/tabular.py (hunk 2 above)
Comment (moezali1) [reply pycaret/internal/tabular.py:68]: Nope

**Review by Yard1 — APPROVED** [review]

**Pull request merged**